micronaut-mcp-server-java-sdk/src/main/java/io/micronaut/mcp/server/registry/AbstractMcpMethodRegistry.java

```diff
@@ -57,6 +107,205 @@ protected final Stream<Method<Object>> drainMethods() {
+        }
+        return new McpError(McpSchema.ErrorCodes.INTERNAL_ERROR);
+    }
+
+    private static Object argumentJsonSchema(Argument<?> argument,
+                                                 Function<Argument<?>, String> argumentDescription) {
+        String description = argumentDescription.apply(argument);
+
+        if (description != null) {
+            Map<String, Object> schema = new LinkedHashMap<>();
+            schema.put(KEY_TYPE, argumentType(argument));
+            if (StringUtils.isNotEmpty(description)) {
+                schema.put(MEMBER_DESCRIPTION, description);
+            }
+            return schema;
+        }
+
+        return new McpSchema.JsonSchema(argumentType(argument), null, null, null, null, null);
+    }
+
+    private static String argumentType(Argument<?> argument) {
+        if (argument.isAssignableFrom(String.class)) {
+            return TYPE_STRING;
+        }
+        return TYPE_OBJECT;
+    }
+
+    private static boolean isArgumentRequired(Argument<?> argument) {
+        return !argument.isNullable();
+    }
+
+    private <B> Object instantiateArgumentViaJsonMapper(ExecutableMethod<B, Object> method,
+                                                        int position,
+                                                        Map<String, Object> arguments) {
+        try {
+            String payload = jsonMapper.writeValueAsString(arguments);
+            Argument<?> argument = method.getArguments()[position];
+            Class<?> classInputSchema = argument.getType();
+            return jsonMapper.readValue(payload, classInputSchema);
+
+        } catch (IOException ex) {
+            throw mcpError(ex);
+        }
+    }
+
+    @Nullable
+    private McpErrorExceptionMapper<? extends Throwable> getExceptionMapper(@NonNull Class<? extends Throwable> exceptionClass) {
+        return classToExceptionMapper.computeIfAbsent(exceptionClass, aClass -> {
+            for (McpErrorExceptionMapper<?> exceptionMapper : exceptionMappers) {
+                if (exceptionMapper.canMap(aClass)) {
+                    return exceptionMapper;
+                }
+            }
+            return null;
+        });
+    }
+
+    protected <B> Object[] methodArgs(ExecutableMethod<B, Object> method,
+                                      Object request,
+                                      Object ctx) {
+        return methodArgs(method, Collections.emptyMap(), request, ctx, Argument::getName);
+    }
+
+    protected <B> Object[] methodArgs(ExecutableMethod<B, Object> method,
+                                    Map<String, Object> arguments,
+                                      Object request,
+                                    Object ctx,
+                                   Function<Argument<?>, String> argumentName) {
+        Object[] args = new Object[method.getArguments().length];
+        LinkedHashMap<Class<?>, Integer> boundArgumentsPositions = boundArgumentsPositions(method);
```

Comment:
seems like you are inventing your own method argument binding logic here. Why not using `DefaultExecutableBinder`? It supports `preBound` types which would handle this use case.